tests/test_edge_cases.py

```diff
@@ -615,6 +615,20 @@ class Config:
     assert str(m) == "Child a=1 b='s'"
 
 
+def test_annotation_inheritance():
+    class A(BaseModel):
+        value: int = 1
+
+    class B(A):
+        value = 'G'
```

Comment:
what happens in the case that I change this to `value: str = 'G'`?

Reply:
It should update the type annotation to `str`, I guess a test case is required.